src/deepsparse/server/build_logger.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Specifies the mapping from the ServerConfig to the DeepSparse Logger
+"""
+
+from typing import Any, Dict, List, Optional, Union
+
+from deepsparse import loggers as logger_objects
+from deepsparse.loggers.config import MetricFunctionConfig, TargetLoggingConfig
+from deepsparse.server.config import ServerConfig
+
+
+__all__ = ["build_logger"]
+
+SUPPORTED_LOGGER_NAMES = ["python"]
+
+
+def build_logger(server_config: ServerConfig) -> logger_objects.BaseLogger:
+    """
+    Builds a DeepSparse logger from the ServerConfig.
+
+    The process follows the following tree-like hierarchy:
+
+    First: the "leaf" loggers are being built.
+    Second: if specified in ServerConfig (key "data_logging"),
+        every "leaf" logger is wrapped inside a FunctionLogger
+    Third: if required, the list of resulting loggers is wrapped inside a
+        MultiLogger
+    Fourth: if specified, the resulting logger is wrapped inside the SubprocessLogger
+
+    :param server_config: the Server configuration model
+    :return: a DeepSparse logger instance
+    """
+
+    loggers_config = server_config.loggers
+    if loggers_config is None:
+        return loggers_config
+
+    leaf_loggers = build_leaf_loggers(loggers_config)
+    loggers = []
+
+    for name, leaf_logger in leaf_loggers.items():
+        target_logging_configs = _get_target_logging_configs(name, server_config)
+
+        if target_logging_configs:
+            logger = logger_objects.FunctionLogger(
+                logger=leaf_logger, target_logging_configs=target_logging_configs
+            )
+        else:
+            logger = leaf_logger
+
+        loggers.append(logger)
+
+    return logger_objects.MultiLogger(loggers) if len(loggers) > 1 else loggers[0]
+
+
+def build_leaf_loggers(
+    loggers_config: List[Union[str, Dict[str, Dict[str, Any]]]]
```

Comment:
I think the typing is a bit off here - `List[Dict[str, ...]]` wouldn't make much sense if the dict keys are logger names, that case instead should be just a dictionary where each key is a logger class and the args are their kwargs